src/components/common/Tabs/EditProfileTab/index.js

```diff
@@ -0,0 +1,196 @@
+import { classNames } from 'primereact/utils';
+import './styles.scss';
+
+const EditProfileTab = () => {
+  const dispatch = useDispatch();
+  const t = useTranslation();
+  const { user } = useSelector(selectAuth);
+  const [changePassword] = useChangePasswordMutation();
+  const [updateUser] = useUpdateUserMutation();
+
+  const onBack = () => {
+    dispatch(setHomeTab(tabsEnum.profile));
+  };
+
+  const onKeyDown = callback => event => {
+    if (event.charCode === 13) callback();
+  };
+
+  const onDeleteAccount = () => {
+    dispatch(error(t('global.notImplemented')));
+  };
+
+  const defaultValues = {
+    current_password: '',
+    password: '',
+    password_confirmation: '',
+    email: user.email,
+  };
+
+  const {
+    control,
+    getValues,
+    handleSubmit,
+    formState: { errors },
+  } = useForm({ defaultValues });
+
+  const onSaveChanges = data => {
+    return Promise.all([
+      changePassword({
+        current_password: data.current_password,
+        password: data.password,
+        password_confirmation: data.password_confirmation,
+      }),
+      updateUser({ id: user.id, user: { ...user, email: data.email } }),
+    ])
+      .then(() => {
+        dispatch(success(t('profile.editSuccess')));
+        dispatch(setHomeTab(tabsEnum.profile));
+      })
+      .catch(() => {
+        dispatch(error(t('profile.editError')));
+      });
+  };
+
+  const getFormErrorMessage = name => {
+    return errors[name] && <small className="p-error">{errors[name].message}</small>;
+  };
+
+  return (
+    <>
+      <div className="header blue">
+        <Back onBack={onBack} />
+        <div className="create-title">{t('editProfile.title')}</div>
+      </div>
+      <div className="profile-picture">
+        <img src={user?.avatar.url || './default-profile.png'} alt="profile" />
+      </div>
+      <div className="profile-name">{user?.email}</div>
+      <div className="create-target">
+        <form className="p-fluid grid formgrid" onSubmit={handleSubmit(onSaveChanges)} noValidate>
+          <div className="field col-12 centered">
+            <label htmlFor="email" className="uppercase edit-profile-label">
+              {t('editProfile.email')}
+            </label>
+            <Controller
+              name="email"
+              control={control}
+              rules={{
+                required: 'Email is required.',
+                pattern: {
+                  value: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i,
+                  message: 'Invalid email address. E.g. example@email.com',
+                },
+              }}
+              render={({ field, fieldState }) => (
+                <InputText
+                  id={field.name}
+                  {...field}
+                  className={classNames('w-10 p-inputtext-sm', { 'p-invalid': fieldState.invalid })}
+                />
+              )}
+            />
+            {getFormErrorMessage('email')}
+          </div>
```

Comment:
Some of the JSX here seems to repeat for each field.
```
<div>...
<label />
<Controller />
{getError}
```
Maybe it can be abstracted so the component ends up being shorter. For example create an array of fields that contains with the _particular things_, and then do a `fields.map` with the _common_ JSX